Bobjayafam/politico
# Ft user signup 163637896

#### What does this PR do?
implement a signup endpoint that allows users to create an account
#### Description of Task to be completed?
- add a user controller class
- add a create user method to the user controller class
- update the connection string to be dynamic to account for different environments
- add a utility function that generates token
- add a utility function for hashing passwords with bcrypt
- include a middleware to validate user data
- update package.json by setting node environment in the test and pretest scripts
- write passing tests for the endpoint
#### How should this be manually tested?
- clone the GitHub repo
- checkout branch ``ft-user-signup-163637896``
- run npm install
- start server with npm run dev
- on POSTMAN, send a POST request to ``/api/v1/auth/signup``.  send a request body which includes firstname, lastname, othername, email, password(6 or more characters), phone number(11 digits) and passportUrl. The response should be a body containing the generated token and the created user object
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163637896
![create-user-db](https://user-images.githubusercontent.com/19153324/52131260-9e76f480-263c-11e9-8b9a-2be88be094da.png)
